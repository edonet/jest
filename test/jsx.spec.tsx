/**
 *****************************************
 * Created by edonet@163.com
 * Created on 2022-12-11 22:13:50
 *****************************************
 */
'use strict';


/**
 *****************************************
 * 加载依赖
 *****************************************
 */
import { describe, test, expect } from '@jest/globals';
import { create } from 'react-test-renderer';


/**
 *****************************************
 * 组件
 *****************************************
 */
function Link(props: { page: string, children: string }) {
  return (
    <a href={props.page || '#'}>
      {props.children}
    </a>
  );
}


/**
 *****************************************
 * 测试
 *****************************************
 */
describe('jsx', () => {
  test('Link', () => {
    const link = create(<Link page="http://www.facebook.com">Facebook</Link>);
    const json = link.toJSON();

    // 校验快照
    expect(json).toMatchSnapshot();
  });
});
