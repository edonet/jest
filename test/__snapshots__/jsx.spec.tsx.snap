// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsx Link 1`] = `
<a
  href="http://www.facebook.com"
>
  Facebook
</a>
`;
